lib/ArtifactUtils.cpp

```diff
@@ -204,7 +204,7 @@ DLL_LINKAGE CArtifactInstance * ArtifactUtils::createArtifact(CMap * map, const
 			art = ArtifactUtils::createScroll(SpellID(spellID));
 		}
 	}
-	else //TODO: create combined artifact instance for random artifacts, just in case
```

Comment:
Generation of combined artifacts on random maps does not match the behavior of OH3. I suggest removing this TODO.

Reply:
Perhaps add an assertion then? So if somebody uses this function for combined artifact he won't end up with broken state.

Reply:
Just checked. If allow in the generator, the combined artifacts will appear on the map. So, assertion is not needed.